crates/cargo-platform/src/check_cfg.rs

```diff
@@ -0,0 +1,71 @@
+//! check-cfg
+
+use std::collections::{HashMap, HashSet};
+
+/// Check Config (aka `--print=check-cfg` representation)
+#[derive(Debug, Default, Clone)]
+pub struct CheckCfg {
+    /// Is `--check-cfg` activated
+    pub exhaustive: bool,
+    /// List of expected cfgs
+    pub expecteds: HashMap<String, ExpectedValues>,
+}
+
+/// List of expected check-cfg values
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ExpectedValues {
+    /// List of expected values
+    Some(HashSet<Option<String>>),
+    /// All values expected
+    Any,
+}
+
+impl CheckCfg {
+    /// Process an output line
+    pub fn process_line(&mut self, line: &str) {
+        if line == "any()=any()" || line == "any()" {
+            self.exhaustive = false;
+            return;
+        }
+
+        let mut value: HashSet<Option<String>> = HashSet::default();
+        let mut value_any_specified = false;
+        let name: String;
+
+        if let Some((n, val)) = line.split_once('=') {
+            name = n.to_string();
+
+            if val == "any()" {
+                value_any_specified = true;
+            } else if val.is_empty() {
+                // no value to add
+                // strip "" around the value, e.g. "linux" -> linux
+            } else if let Some(val) = val.strip_prefix('"').map(|v| v.strip_suffix('"')).flatten() {
+                value.insert(Some(val.to_string()));
+            } else {
+                // missing quotes and non-empty, ignore
+                return;
+            }
+        } else {
+            name = line.to_string();
+            value.insert(None);
+        }
+
+        self.expecteds
+            .entry(name)
+            .and_modify(|v| match v {
+                ExpectedValues::Some(v) if !value_any_specified => {
+                    v.extend(value.clone());
+                }
+                ExpectedValues::Some(_) => *v = ExpectedValues::Any,
```

Comment:
Generally people find it easier to read positive logic over negative logic.  If you just swapped these two lines, I think you can remove `!` and have the same result.